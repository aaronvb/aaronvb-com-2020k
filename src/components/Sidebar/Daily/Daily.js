// @flow strict
import React from 'react';
import styles from './Daily.module.scss';

type Props = {
  dayOfWeek: number
};

const getWeekDay = (dayOfWeek) => {
  let msg;
  switch (dayOfWeek) {
    case 0:
      msg = 'Happy Weekend, rest well. 😴';
      break;
    case 1:
      msg = 'Happy Monday. ☕☕☕';
      break;
    case 2:
      msg = 'Happy Tuesday. ☕';
      break;
    case 3:
      msg = 'Happy Wednesday. ☕';
      break;
    case 4:
      msg = 'Happy Thursday. ☕';
      break;
    case 5:
      msg = 'Happy Aloha Friday. 🏄';
      break;
    case 6:
      msg = 'Happy Weekend, enjoy! 🍺';
      break;
    default:
      msg = '';
  }
  return msg;
};

const Copyright = ({ dayOfWeek }: Props) => (
  <div className={styles['daily']}>
    {getWeekDay(dayOfWeek)}
  </div>
);

export default Copyright;
