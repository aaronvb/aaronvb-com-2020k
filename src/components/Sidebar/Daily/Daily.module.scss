@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.daily {
  @include margin-bottom(.25);
  color: lighten($color-gray, 18%);
  font-size: $typographic-small-font-size;
}