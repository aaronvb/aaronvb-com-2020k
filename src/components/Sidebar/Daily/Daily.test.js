// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Daily from './Daily';
import MockDate from 'mockdate';

const getCurrentDate = () => new Date();
describe('Daily', () => {
  it('renders correctly', () => {
    MockDate.set('2020-04-30')
    const tree = renderer.create(<Daily />).toJSON();
    expect(tree).toMatchSnapshot();
    MockDate.reset();
  });
});
