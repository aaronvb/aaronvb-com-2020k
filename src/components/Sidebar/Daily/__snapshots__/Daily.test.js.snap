// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daily renders correctly 1`] = `
<div
  className="daily"
>
  Happy Wednesday. â˜•
</div>
`;
